src/SpaceService.ts

```diff
@@ -169,7 +237,15 @@ class SpaceService extends Base {
     const { apiEndpointAddress, got } = this.lensPlatformClient;
     const url = `${apiEndpointAddress}/spaces/${cluster.space?.name}/k8sclusters/${cluster.id}`;
 
-    const json = await got.put(url, { json: cluster });
+    const json = await throwExpected(
+      () => got.put(url, { json: cluster }),
+      {
+        // eslint-disable-next-line
+        // TODO: differentiate between space and cluster not being found
+        404: () => new SpaceNotFoundException(cluster.space?.name ?? "undefined"),
+        400: () => new BadRequestException("Property 'kind' of cluster object is invalid")
```

Comment:
I think there are multiple reasons for the bad request, not just wrong kind.

Reply:
Certainly, there are a bunch of places where existing error handling needs improvements and testing. I think it would make sense to do this in subsequent PRs.

Reply:
Yes, cluster functions still need some work. I was thinking of delegating this part to subsequent PRs